pax-exam-container-carbon/org.wso2.carbon.container/src/main/java/org/wso2/carbon/container/ArchiveExtractor.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.ops4j.pax.exam.TestContainerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.file.Path;
+
+/**
+ * Extract zip or tar.gz archives to a target Directory.
+ */
+public class ArchiveExtractor {
+
+    private static final Logger logger = LoggerFactory.getLogger(ArchiveExtractor.class);
+
+    private ArchiveExtractor() {
+    }
+
+    /**
+     * Extract zip or tar.gz specified by maven url to a target Directory.
+     *
+     * @param sourceURL       url of the archive to extract
+     * @param targetDirectory where to extract to
+     * @throws IOException on I/O error
+     */
+    public static void extract(URL sourceURL, File targetDirectory) throws IOException {
+        if (sourceURL.getProtocol().equals("file")) {
+            String file = sourceURL.getFile();
+            if (file.endsWith(".zip")) {
+                extractZipDistribution(sourceURL, targetDirectory);
+            } else if (file.endsWith(".tar.gz")) {
+                extractTarGzDistribution(sourceURL, targetDirectory);
+            } else {
+                throw new TestContainerException(
+                        "Unknown packaging of distribution; only zip or tar.gz could be handled.");
+            }
+            return;
+        }
+        if (sourceURL.toExternalForm().endsWith("/zip")) {
+            extractZipDistribution(sourceURL, targetDirectory);
+        } else if (sourceURL.toExternalForm().endsWith("/tar.gz")) {
+            extractTarGzDistribution(sourceURL, targetDirectory);
+        } else {
+            throw new TestContainerException(
+                    "Unknown packaging; only zip or tar.gz could be handled. URL was " + sourceURL);
+        }
+    }
+
+    /**
+     * Extract zip file in the system to a target Directory.
+     *
+     * @param path            path of the archive to be extract
+     * @param targetDirectory where to extract to
+     * @throws IOException on I/O error
+     */
+    public static void extract(Path path, File targetDirectory) throws IOException {
+        if (path.toString().endsWith(".zip")) {
+            extract(new ZipArchiveInputStream(new FileInputStream(path.toFile())), targetDirectory);
+        } else {
+            throw new TestContainerException("Unknown packaging of distribution; only zip can be handled.");
+        }
+    }
+
+    private static void extractTarGzDistribution(URL sourceDistribution, File targetDirectory) throws IOException {
```

Comment:
@Niranjan-K : If method name self describe what method does, Still do we have to add javadocs for private? IMO it is unnecessary